### DISCUSSION TITLE
Simulation of coaxial cable

### DISCUSSION THREAD

**Discussion by GMellar** [thliebig/openEMS-Project#242] [Q&A Usage]
I have designed a simple test case for a coaxial cable. The boundary condition is PML and there is a lumped port to feed the cable. I need to simulate a 50Ohm cable so I need to feed it with a lumped port. At the moment there is no coaxial port with a resistance more than 0 implemented in openEMS. I put the port inside the coaxial cable but I don't know if that is a good idea. With a PML boundary condition I would expect there is not much reflected but the result gives me noticable reflection. I think this is the port an how I place it. My question is how to setup the excitation for that cable properly. 

I attached all files of the simulation to reproduce it. 

[sim.zip](https://github.com/user-attachments/files/17508199/sim.zip)

**Comment by VolkerMuehlhaus** [comment]
Your line touches the PML wall at both ends, so your port resistance is in parallel to the absorbing wall. You need to add some spacing (air gap).

<img width="419" alt="coax" src="https://github.com/user-attachments/assets/967322a9-6f2a-4284-beb5-8b65df1edffb">

That said, isn't there an optional parameter Feed_R in the coax port definition?
https://github.com/thliebig/openEMS/blob/master/matlab/AddCoaxialPort.m

**Reply by GMellar** [reply]
> Your line touches the PML wall at both ends, so your port resistance is in parallel to the absorbing wall. You need to add some spacing (air gap).
> <img alt="coax" width="419" src="https://private-user-images.githubusercontent.com/47922981/379907951-967322a9-6f2a-4284-beb5-8b65df1edffb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4Mzc0NDcsIm5iZiI6MTcyOTgzNzE0NywicGF0aCI6Ii80NzkyMjk4MS8zNzk5MDc5NTEtOTY3MzIyYTktNmYyYS00Mjg0LWJlYjUtOGI2NWRmMWVkZmZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDA2MTkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlNjY1ZmIxODM4YzE5NTEwNjEyMDdmOGYyYmU2NjAyOTg1MjVhMjQzZDg4YTYzMmQ2MDBlMDgzYjE4MWRkMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TefJ0TWrx702Xl3B5U-Ml1XLICAEKvG-PQ3DD-z-4Fc">
> 
> That said, isn't there an optional parameter Feed_R in the coax port definition? https://github.com/thliebig/openEMS/blob/master/matlab/AddCoaxialPort.m

There is an optional parameter for a feed R but this is not implemented at the moment so any value larger zero will not work with that kind of port. 

I thought the idea of PML is to match the end of the line but I think you mean the start of the line must not touch the PML wall. I tested that with a distance of 1mm to the wall at the beginning of the line. The resulting reflection is around -21dB which is better than before and the calculated Z_abs is around 42Ohm which is lower than I expected but I getting near the target impedance. 

[sim.zip](https://github.com/user-attachments/files/17517043/sim.zip)

**Reply by Miceuz** [reply]
Not to hijack the thread, but I think my question is related. Maybe @VolkerMuehlhaus can shine some light on this issue?

How do I use PML as a terminator and how to avoid PML in normal simulations?

PML_8 is said to use up 8 cells from edge of simbox if I want to use it as terminator, do I make my simbox so that my metal is 8 cells away from edge or do I use PML_1 and put edge of simbox right at the edge of metal or what?

 I have read somewhere on this forum, that only PML_8 ir proven, different distances are experimental, is it true?

Similarly, if I want to use PML_8 as normal absorbing boundary, it is said that I need $\lambda$/4 space to boundary. Is it $\lambda$/4 + 8 cells to simbox edge or are 8 cells counted into $\lambda$/4 distance?

This thread is useful for the topic - https://github.com/thliebig/openEMS-Project/discussions/159

**Reply by VolkerMuehlhaus** [reply]
> I thought the idea of PML is to match the end of the line but I think you mean the start of the line must not touch the PML wall

The point is: you can terminate the port by running it into a PML wall *OR* by an internal resistance. But don't use both at the same time!

The PML will absorb signals travelling along the line, no matter what the line impedance is, so that can be useful if you build a line with a port for excitation and then run it into the PML wall (no second port used here). The input impedance into the line is then equal to Zline, because there is no reflected signal at the end of the line, due to the PML.

If you simulate the line with ports on both ends and have internal resistor (not running into PML), you will get the true physical behavious that you see in the lab for lines terminated by some fixed impedance.

**Reply by GMellar** [reply]
> > I thought the idea of PML is to match the end of the line but I think you mean the start of the line must not touch the PML wall
> 
> The point is: you can terminate the port by running it into a PML wall _OR_ by an internal resistance. But don't use both at the same time!
> 
> The PML will absorb signals travelling along the line, no matter what the line impedance is, so that can be useful if you build a line with a port for excitation and then run it into the PML wall (not second port used here). The input impedance into the line is then equal to the Zline, because there is no reflected signal at the end of the line, due to the PML.
> 
> If you simulate the line with ports on both ends and have internal resistor (not running into PML), you will get the true physical behavious that you see in the lab for lines terminated by some fixed impedance.

In this case I have an internal resistance at the exciting port and the line ends in a PML wall which should result in an input impedance of ZLine, is that right?

I experimented a bit with the PML wall and I think the distance to my exciting port was messing things up. What I did was to use PML only on the wall where the coaxial line ends. I set the other walls to PEC which should be no problem in this case because most of the field is inside the coax line. I attached  the simulation files and the result is promising because the reflection is much lower and the measured line impedance is near 50Ohm now. 

[sim.zip](https://github.com/user-attachments/files/17522116/sim.zip)

**Reply by VolkerMuehlhaus** [reply]
> In this case I have an internal resistance at the exciting port and the line ends in a PML wall which should result in an input impedance of ZLine, is that right?

Yes, correct. 

One comment on your lumped port to drive the line: I would make it wider, same width as the center conductor, to minimize parasitic inductance in the port. 

<img width="655" alt="coax_2" src="https://github.com/user-attachments/assets/7e712955-e1ef-4815-b97f-4d6f047291d6">

How many mesh cells from PML to create a gap: I have no insight what is the minimum mesh cell count from the PML wall to really create a gap there - these implementation details are different between openEMS and my commercial FDTD that I use at work. Hopefully someone else can answer that.

**Comment by Miceuz** [comment]
I am afraid resistance >0 is not supported yet:


```
if (feed_R > 0) && ~isinf(feed_R)
    error 'feed_R not yet implemented
```
